cmake/Buildmatlab-whole-body-simulator.cmake

```diff
@@ -0,0 +1,34 @@
+# Copyright (C) Fondazione Istituto Italiano di Tecnologia
+# CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+include(YCMEPHelper)
+include(FindOrBuildPackage)
+
+find_or_build_package(YARP QUIET)
+find_or_build_package(yarp-matlab-bindings QUIET)
+find_or_build_package(WBToolbox QUIET)
+find_or_build_package(iDynTree QUIET)
+find_or_build_package(qpOASES QUIET)
+find_or_build_package(icub-models QUIET)
+find_or_build_package(whole-body-controllers QUIET)
+
+set(matlab-whole-body-simulator_DEPENDS "")
+list(APPEND matlab-whole-body-simulator_DEPENDS YARP)
+list(APPEND matlab-whole-body-simulator_DEPENDS yarp-matlab-bindings)
+list(APPEND matlab-whole-body-simulator_DEPENDS WBToolbox)
+list(APPEND matlab-whole-body-simulator_DEPENDS iDynTree)
+list(APPEND matlab-whole-body-simulator_DEPENDS qpOASES)
```

Comment:
There is no need for including explicitly all the transitive dependencies of WB-Toolbox such as iDynTree or qpOASES, listing WBToolbox as a dependency is enough (same goes for YARP, if we list yarp-matlab-bindings as a dependency it is enough).

Reply:
I guess the same goes for the `find_or_build_package(...)` instruction lines, right?

Reply:
Yes, I forgot about those, thanks!

Reply:
I would prefer to keep the dependency qpOASES, as in a near future, WBToolbox could replace qpOASES with OSQP, and this could happen before the **matlab-whole-body-simulator** is updated. In such case it's better to leave the check in order to warn the user.

Reply:
As discussed on https://github.com/robotology/robotology-superbuild/issues/239#issuecomment-774995229, qpOASES will remain in the superbuild and there is no plan to remove it. I think it is safe to remove from here and just leave it in the WBToolbox dependencies.

Reply:
(even because the fact that qpOASES is found in the system does not ensure you that WBToolbox was compiled with qpOASES support, that it was you actually need in the `matlab-whole-body-simulator` .

Reply:
ok, perfect.

Reply:
Done in https://github.com/robotology/robotology-superbuild/pull/689/commits/b24f8ee65e8a7c741019a015316c46474cc356b2.